scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -886,10 +925,15 @@ void BPF_STRUCT_OPS(layered_dispatch, s32 cpu, struct task_struct *prev)
 
 	/* consume !preempting open layers */
 	bpf_for(idx, 0, nr_layers) {
+		layer_idx = iter_layer_cpu_ctx(cctx->layer_idx, idx);
+		if (layer_idx > MAX_LAYERS) {
```

Comment:
I tried to drop these checks but was hitting verifier errors.

Reply:
Does shifting the test into `iter_layer_cpu_ctx()` before returning work?

Reply:
Yeah, let me try to clean that up.